test/e2e/lib/mocha-hooks.js

```diff
@@ -51,8 +52,51 @@ afterEach( function () {
 	}
 } );
 
-// Take Screenshot
+// Take Screenshot for Playwright.
 afterEach( async function () {
+	if ( ! global.isPlaywright ) {
+		return;
+	}
+
+	this.timeout( afterHookTimeoutMS );
+
+	// If no test has been run, don't bother.
+	if ( ! this.currentTest ) {
+		return;
+	}
+	// If option is set to never save screenshots, then don't bother.
+	const doNotSaveScreenshot = config.get( 'neverSaveScreenshots' );
+	if ( doNotSaveScreenshot ) {
+		return;
+	}
+	// If test is passed and option to always save screenshot is not set,
+	// don't bother.
+	const saveAllScreenshot = config.get( 'saveAllScreenshots' );
+	if ( this.currentTest.state === 'passed' && ! saveAllScreenshot ) {
+		return;
+	}
+
+	// Now let's get to work saving the screenshot.
+	// Retrieve the failing page, obtain the test title, viewport dimensions, timestamp.
+	// Then call Playwright's built-in screenshot utility and save it as PNG.
+	const page = this.currentTest.page;
+	const shortTestFileName = this.currentTest.title.replace( /[^a-z0-9]/gi, '-' ).toLowerCase();
+	const screenSize = browserManager.targetScreenSize().toUpperCase();
+	const locale = browserManager.targetLocale().toUpperCase();
+	const date = new Date().getTime().toString();
+	const fileName = `FAILED-${ locale }-${ screenSize }-${ shortTestFileName }-${ date }`;
```

Comment:
Shouldn't `FAILED` be dynamic and depend on `this.currentTest.state`? Otherwise, when `saveAllScreenshots:true` all screenshots will be `FAILED-*.png`, even for passing tests.